readme.md

```diff
@@ -7,67 +7,257 @@
+)
 ```
 
-This will produce a binary `septentrio_example` which can be executed.
+Then you can add `@p1_polaris//c:polaris_client` to the `deps` section of a `cc_binary()` or `cc_library()` rule in your
+project, and add `#include <point_one/polaris/polaris.h>` to your source code. For example:
+
+```bazel
+cc_binary(
+    name = "my_application",
+    srcs = ["main.c"],
+    deps = ["@p1_polaris//c:polaris_client"],
+)
 ```
-# Get list of arguments
-./septentrio_example --help
+
+> Note that you do not need to clone the Polaris repository when using Bazel. Bazel will clone it automatically when you
+build your application.
+
+##### Compiling Example Applications With Bazel ####
+
+To compile and run the included example applications using Bazel, follow these steps:
+
+1. Clone the Polaris source code and navigate to the `c/` source directory:
+   ```bash
+   git clone https://github.com/PointOneNav/polaris.git
+   cd polaris
+   ```
+2. Compile the Polaris source code and example applications:
+   ```bash
+   bazel build -c opt //c/examples:*
+   ```
+
+#### CMake ####
+
+> Note: The C client does not currently have any external dependencies.
+
+1. Clone the Polaris source code and navigate to the `c/` source directory:
+   ```bash
+   git clone https://github.com/PointOneNav/polaris.git
+   cd polaris/c
+   ```
+2. Create a `build/` directory and run CMake to configure the build tree:
+   ```bash
+   mkdir build
+   cd build
+   cmake ..
+   ```
+3. Compile the Polaris source code and example applications:
+   ```bash
+   make
+   ```
+
+The generated example applications will be located in `build/examples/<APPLICATION NAME>`.
+
+#### GNU Make ###
+
+> Note: The C client does not currently have any external dependencies.
+
+1. Clone the Polaris source code and navigate to the `c/` source directory:
+   ```bash
+   git clone https://github.com/PointOneNav/polaris.git
+   cd polaris/c
+   ```
+3. Compile the Polaris source code and example applications:
+   ```bash
+   make
+   ```
+
+The generated example applications will be located in `examples/<APPLICATION NAME>`.
+
+### Using Polaris C Client ###
+
+1. Include `polaris.h`:
+   ```c
+   #include <point_one/polaris/polaris.h>
+   ```
+2. Create a `PolarisContext_t` context object, which contains the data send/receive buffers, socket for communicating
+   with Polaris, and other configuration details.
+   ```c
+   PolarisContext_t context;
+   Polaris_Init(&context);
+   ```
+3. Authenticate with Polaris, providing your assigned API key and a unique ID for this connection instance. This will
+   generate an _authentication token_, which will be used to receive corrections data.
+   ```c
+   Polaris_Authenticate(&context, api_key, unique_id);
+   ```
+   - Note that the ID you specify must be unique across all Polaris sessions using your API key. If two connections use
+     the same API key, they will conflict and will not work correctly.
```

Comment:
i think you meant to say if two connections use the same Unique ID. Two can use the same API key no problem.

Reply:
Yes, typo